app/routes.py

```diff
@@ -1,2 +1,110 @@
-from flask import Blueprint
+from app import db
+from app.models.planet import Planet
+from flask import Blueprint, jsonify, make_response, request
 
+# path = "https://api.le-systeme-solaire.net/rest.php/bodies"
+
+
+planets_bp = Blueprint("planets", __name__, url_prefix="/planets")
+
+
+@planets_bp.route("", methods=["GET", "POST"])
+def handle_planets():
+    if request.method == "GET":
+        name = request.args.get("name")
+        if name:
+            planets = Planet.query.filter_by(name=name) 
+        else:
+            planets = Planet.query.all()
+        planets_response = []
+        for planet in planets:
+            planets_response.append(
+                {
+                    "id": planet.id,
+                    "name": planet.name,
+                    "description": planet.description,
+                    "color": planet.color
+                }
+            )
+        return jsonify(planets_response), 200
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        new_planet = Planet(name=request_body["name"],
+                            description=request_body["description"],
+                            color=request_body["color"]
+                            )
+
+        db.session.add(new_planet)
+        db.session.commit()
+
+        new_planet_response = { 
+            "id": new_planet.id,
+            "name": new_planet.name,
+            "description": new_planet.description,
+            "color": new_planet.color
+        }
+
+        return jsonify(new_planet_response), 201
+
+
+@planets_bp.route("/<planet_id>", methods=["GET", "PUT", "DELETE"])
```

Comment:
Consider separating each request method into its different functions. This makes testing the endpoints much easier to do and follows the single responsibility principle.

Reply:
The same can be said about the `"/planets"` endpoint too